ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,254 @@
+    """
+
+    def __init__(
+        self, project_id, client=None,
+    ):
+        self.client = client or TraceServiceClient()
+        self.project_id = project_id
+
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        """Export the spans to Cloud Trace.
+
+        See: https://cloud.google.com/trace/docs/reference/v2/rest/v2/
+             projects.traces/batchWrite
+
+        Args:
+            spans: Tuple of spans to export
+        """
+        cloud_trace_spans = []
+        for span in self.translate_to_cloud_trace(spans):
+            try:
+                cloud_trace_spans.append(self.client.create_span(**span))
+            # pylint: disable=broad-except
+            except Exception as ex:
+                logger.warning("Error %s when creating span %s", ex, span)
+
+        try:
+            self.client.batch_write_spans(
+                "projects/{}".format(self.project_id), cloud_trace_spans,
+            )
+        # pylint: disable=broad-except
+        except Exception as ex:
+            logger.warning("Error while writing to Cloud Trace: %s", ex)
+            return SpanExportResult.FAILURE
+
+        return SpanExportResult.SUCCESS
+
+    def translate_to_cloud_trace(
+        self, spans: Sequence[Span]
+    ) -> List[Dict[str, Any]]:
+        """Translate the spans to Cloud Trace format.
+
+        Args:
+            spans: Tuple of spans to convert
+        """
+
+        cloud_trace_spans = []
+
+        for span in spans:
+            ctx = span.get_context()
+            trace_id = _get_hexidecimal_trace_id(ctx.trace_id)
+            span_id = _get_hexidecimal_span_id(ctx.span_id)
+            span_name = "projects/{}/traces/{}/spans/{}".format(
+                self.project_id, trace_id, span_id
+            )
+
+            parent_id = None
+            if isinstance(span.parent, trace_api.Span):
+                parent_id = _get_hexidecimal_span_id(
+                    span.parent.get_context().span_id
+                )
+            elif isinstance(span.parent, trace_api.SpanContext):
+                parent_id = _get_hexidecimal_span_id(span.parent.span_id)
+
+            start_time = get_time_from_ns(span.start_time)
+            end_time = get_time_from_ns(span.end_time)
+
+            attributes = _extract_attributes(span.attributes)
+
+            cloud_trace_spans.append(
+                {
+                    "name": span_name,
+                    "span_id": span_id,
+                    "display_name": get_truncatable_str_object(span.name),
+                    "start_time": start_time,
+                    "end_time": end_time,
+                    "parent_span_id": parent_id,
+                    "attributes": attributes,
+                    "links": _extract_links(span.links),
+                    "status": _extract_status(span.status),
+                    "time_events": _extract_events(span.events),
+                }
```

Comment:
I think we actually could get some of the stackdriver-specific span attrs here: `same_process_as_parent_span`, `child_span_count`, but that's a problem for another PR.

Reply:
Added TODO